### PR TITLE
add `zipWithIndex` to `Process1`

### PR DESCRIPTION
**Pull request by ShengC**
I took the liberty of adding `zipWithIndex` method to `Chunk` as well. If you feel there is better place to hold it, leave me a message. Thanks!

### CODE REVIEW

File: src/main/scala/fs2/Process1.scala [hunk 1]
```diff
@@ -94,6 +94,18 @@ object process1 {
...
       
+  /** Zip the elements of the input `Handle `with its indices, and return the new `Handle` */    
+  def zipWithIndex[F[_], I](implicit F: NotNothing[F]): Handle[F, I] => Pull[F, (I, Int), Handle[F, I]] = {
+    def go(n: Int)(h: Handle[F, I]): Pull[F, (I, Int), Handle[F, I]] = {
+      Pull.await(h) flatMap {
+        case chunk #: h =>
+          Pull.output(chunk.zipWithIndex.map({ case (c, i) => (c, i + n) })) >> go(n + chunk.size)(h)
+      }
+    }
+
```
Comment (pchiusano) [inline src/main/scala/fs2/Process1.scala]: No need for line break here.

**Comment by pchiusano** [comment]
Let's wait for @fthomas to merge his receive functions, then you can implement in terms of those.

**Comment by ShengC** [comment]
Ok, sounds good.

File: src/main/scala/fs2/Process1.scala [hunk 2]
```diff
@@ -94,6 +94,17 @@ object process1 {
           }
       }
       
+  /** Zip the elements of the input `Handle `with its indices, and return the new `Handle` */    
+  def zipWithIndex[F[_], I](implicit F: NotNothing[F]): Handle[F, I] => Pull[F, (I, Int), Handle[F, I]] = {
+    def go(n: Int): Handle[F, I] => Pull[F, (I, Int), Handle[F, I]] = {
+      Pull.receive({
```
Comment (pchiusano) [inline src/main/scala/fs2/Process1.scala]: No need for the parents here, the braces are sufficient. Pull.receive { case chunk #: h => ... } all on one line.

**Comment by pchiusano** [comment]
Can you also add an alias to ProcessOps? Follow the existing conventions there.

**Comment by ShengC** [comment]
@pchiusano does it look good to merge now ?

**Comment by pchiusano** [comment]
Looks good.

Actually I'll probably reformat the code a bit after but this is fine to merge. :) I should produce a style guide at some point.

Thanks!

**Pull request merged**